### ISSUE TITLE
Training strategy?

### ISSUE THREAD

**Issue by zhangzibin** [LantaoYu/SeqGAN#18]
Hi, thanks for your excellent work.
I found your code train discriminator for 15 epoches for each step of generator. It seems is totally an unfair strategy. How do you choose this strategy?

**Comment by LantaoYu** [comment]
Theoretically, in generative adversarial networks, the generator can perfectly fit the underlying data distribution only when the discriminator is optimal while we are updating the generator. Another way to see it is that GANs is a minimax game, as introduced in the Eq. 1 of the paper (https://arxiv.org/pdf/1406.2661.pdf), but in implementation we kind of optimize the loss function in a (min-max and max-min) iteration, note that min-max and max-min are different and to approximate the min-max training scheme, each time we need to train the discriminator as fully as possible. However, there is always a tradeoff between the training time and the performance. In a nutshell, to guarantee the convergence, we normally need to train the discriminator more sufficiently.
